processing/src/main/java/org/apache/carbondata/processing/newflow/sort/Sorter.java

```diff
@@ -39,11 +39,13 @@
    * Sorts the data of all iterators, this iterators can be
    * read parallely depends on implementation.
    *
-   * @param iterators array of iterators to read data.
    * @return
    * @throws CarbonDataLoadingException
    */
-  Iterator<CarbonRowBatch>[] sort(Iterator<CarbonRowBatch>[] iterators)
+  Iterator<CarbonRowBatch>[] sort()
+      throws CarbonDataLoadingException;
+
+  void prepare(Iterator<CarbonRowBatch>[] iterators)
```

Comment:
Already previous steps file resources are closed where and when the use of it is over. I don't think this method is required.